core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala

```diff
@@ -52,6 +54,27 @@ class BarrierTaskContextSuite extends SparkFunSuite with LocalSparkContext {
     assert(times.max - times.min <= 1000)
   }
 
+  test("share messages with allGather() call") {
```

Comment:
Shall we add a extra tests for mix-in barrier & allGather, e.g.:

```
rdd.barrier().mapPartitions { it =>
      val context = BarrierTaskContext.get()
      val partitionId = context.partitionId
      if (partitionId == 0) {
        context.barrier()
      } else {
        context.allGather("partitionId.toString)
      }
```

Reply:
maybe we should add the first one you mentioned, but that can be left as a followup. What do we expect from the second test case ?

Reply:
Just want to make sure that different barrier methods can work well  in the same stage, since they'll share some data in `BarrierCoordinator`.

Reply:
+1. Btw, the first one shall fail.

Reply:
Thanks all! Added